app/components/domains-table.tsx

```diff
@@ -0,0 +1,147 @@
+  IconButton,
+  Flex,
+  Tooltip,
+  Card,
+  useToast,
+} from '@chakra-ui/react';
+import type { Record, RecordStatus } from '@prisma/client';
+import {
+  EditIcon,
+  DeleteIcon,
+  CheckCircleIcon,
+  RepeatIcon,
+  CopyIcon,
+  TimeIcon,
+  WarningIcon,
+} from '@chakra-ui/icons';
+
+interface DomainsTableProps {
+  domains: Record[];
+  onAction: (domaindID: number, action: DomainsTableAction) => void;
+}
+
+export type DomainsTableAction = 'EDIT' | 'DELETE' | 'RENEW';
+
+export default function DomainsTable(props: DomainsTableProps) {
+  const { domains, onAction } = props;
+
+  const toast = useToast();
+
+  function onCopyNameToClipboard(name: string) {
+    navigator.clipboard.writeText(name);
+    toast({
+      title: 'Name was copied to clipboard',
+      position: 'bottom-right',
+      status: 'success',
+    });
+  }
+
+  function renderDomainStatus(action: RecordStatus) {
+    switch (action) {
+      case 'active':
+        return (
+          <Tooltip label="Domain is live">
+            <CheckCircleIcon color="green.500" boxSize="6" />
+          </Tooltip>
+        );
+      case 'error':
+        return (
+          <Tooltip label="Domain error">
+            <WarningIcon color="brand.500" boxSize="6" />
+          </Tooltip>
+        );
+      case 'pending':
+        return (
+          <Tooltip label="Domain is pending">
+            <Flex
+              width="6"
+              height="6"
+              borderRadius="20"
+              alignItems="center"
+              justifyContent="center"
+              background="yellow.500"
+            >
+              <TimeIcon color="white" boxSize="3" />
+            </Flex>
+          </Tooltip>
+        );
+    }
+  }
+
+  return (
+    <Card p="2" mt="4">
+      <TableContainer>
+        <Table variant="striped" colorScheme="gray">
+          <Thead>
+            <Tr>
+              <Th />
+              <Th>Name</Th>
+              <Th>Type</Th>
+              <Th>Value</Th>
+              <Th>Expiration date</Th>
+              <Th />
+            </Tr>
+          </Thead>
+          <Tbody>
+            {domains.map((domain) => {
+              return (
+                <Tr key={domain.id}>
+                  <Td>{renderDomainStatus(domain.status)}</Td>
+                  <Td>
+                    <Flex justifyContent="space-between" alignItems="center">
+                      {domain.name}
+                      <Tooltip label="Copy name to clipboard">
+                        <IconButton
+                          icon={<CopyIcon color="black" boxSize="5" />}
+                          aria-label="Refresh domain"
+                          variant="ghost"
+                          ml="2"
+                          onClick={() => onCopyNameToClipboard(domain.name)}
+                        />
+                      </Tooltip>
+                    </Flex>
+                  </Td>
+                  <Td>{domain.type}</Td>
+                  <Td>{domain.value}</Td>
```

Comment:
Do we have a maxWidth for each column?

Reply:
I think that it is just better to let Chakra decide the size, if the table doesn't cover fit in the screen it just gets horizontally scrollable

Reply:
Can't we set a max character for the domain attributes, and have a tooltip for users to hover so they can view the entire thing? Should make the table look more clean no?

Reply:
Since we have entire page width for a table I don't see a big need to cut something off + value is the kinda thing that users might need to check frequently and this is pretty much one of the most valuable fields, so again don't see a good reason to hide it

Reply:
I'm not saying to hide it, but to lessen the characters to make the table more aligned? A simple '...' at the end should imply that there's more to the text no?